Add mention about usage of `key` attribute in <Transition> (#2213)
vuejs/docs: src/api/built-in-components.md

    <div v-if="ok">toggled content</div>
  </Transition>
  ```
  

>   Transition on changing key attribute:
>   
>   ```vue-html
>   <Transition>
>     <div :key="data-id">{{changing-text}}</div>

Reviewer: The values `data-id` and `changing-text` wouldn't work in practice, as the hyphens would be treated as subtraction operators.

Maybe something like this instead?

```suggestion
  Forcing a transition by changing the `key` attribute:
  
  ```vue-html
  <Transition>
    <div :key="text">{{ text }}</div>
```
